### ISSUE TITLE
Speakerdeck url's are not consumable by jquery-oembed-all

### ISSUE THREAD

**Issue by mattstratton** [devopsdays/devopsdays-theme#508]
I don't recall the specifics, but speakerdeck URL's were not working.

Reproduction steps:

A test talk is at https://dev.devopsdays.org/events/2017-ponyville/program/adam-jacob (you can build it locally to test).

The speakerdeck code was removed from the talk layout, but it could be added similarly:

```
      {{ if isset .Params "speakerdeck" }}
        {{ if ne .Params.speakerdeck ""}}
          <div>
            <h2 class="talk-page">Speakerdeck</h2>
            <a href="{{ .Params.speakerdeck }}" class="embed"></a>
          </div>
        {{ end }}
      {{ end }}
```

**Comment by mattstratton** [comment]
More details: https://github.com/nfl/jquery-oembed-all

**Comment by mattstratton** [comment]
Here are the errors when loading Speakerdeck:

```
devopsdays-min.js:6 Uncaught TypeError: Cannot read property 'json' of null
    at Object.datareturn (devopsdays-min.js:6)
    at Object.success (devopsdays-min.js:6)
    at u (devopsdays-min.js:2)
    at Object.fireWith [as resolveWith] (devopsdays-min.js:2)
    at i (devopsdays-min.js:3)
    at HTMLScriptElement.n (devopsdays-min.js:3)
    at HTMLScriptElement.dispatch (devopsdays-min.js:2)
    at HTMLScriptElement.g.handle (devopsdays-min.js:2)
```

Also this fun one:
```
Access to Font at 'https://public.slidesharecdn.com/fonts/fontawesome-webfont.woff2?v=4.3.0?cb=1494368673' from origin 'https://www.slideshare.net' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'https://www.slideshare.net' is therefore not allowed access.
```

**Comment by mattstratton** [comment]
I have this built in a branch - you can see it here - https://deploy-preview-509--devopsdays-theme.netlify.com/events/2017-ponyville/program/rainbow-dash/

In order to see the issue, open the page in chrome dev tools and look at the javascript console.

**Comment by aaronblythe** [comment]
I am by no means a javascript person.  However I think this has to do with an oddity of oembed and how it returns a json response of what you want. Out of curiosity have you looked at: https://speakerdeck.com/faq#oembed

Specifically:

For more documentation on oEmbed please see oembed.com. Our oEmbed endpoint is:
`http://speakerdeck.com/oembed.json`
so requesting an oEmbed would look like this:
`http://speakerdeck.com/oembed.json?url=URL_OF_THE_PRESENTATION`

So I think what needs to be rendered is:

``http://speakerdeck.com/oembed.json?url=https://speakerdeck.com/mattstratton/shifting-left-secure"

Or something similar.  Testing this now.  This returns json with an 'html' field when doing a simple get in Postman.  This HTML embeds fine in the site when I just copy and paste it in layouts/talk/single.html

My guess from here is that you have to use Hugo to call to get this json and parse out the html field. https://gohugo.io/extras/datadrivencontent/

**Comment by mattstratton** [comment]
I'm wondering if we just need to construct the URL like you said instead of just the bare URL of the presentation.

**Comment by mattstratton** [comment]
I just tried this in https://github.com/devopsdays/devopsdays-theme/pull/509 - let's see.

**Comment by aaronblythe** [comment]
Unfortunately that URL will not work either because it will return the json.
I tried:
```
            {{ $dataJ := getJSON "http://speakerdeck.com/oembed.json?url=https://speakerdeck.com/mattstratton/shifting-left-securely&format=json" }}
            {{ $dataJ.html }}
```
Unfortunately I don't understand Hugo well enough to get this to render HTML instead of a string.

There is also:
http://gohugo.io/extras/shortcodes/#speaker-deck

Where you just give it the data-id, which is not part of the URL.

**Comment by mattstratton** [comment]
The problem is this should actually WORK using the jquery library. It's how the SlideShare ones work. 

I don't want to have people have to figure out more than they need to; 

Hugo isn't doing any lifting here; I was doing it all in jquery.

**Comment by mattstratton** [comment]
We can't really use short codes in a template; but if people want to use that shortcode they are welcome to do so. I want to basically have them give us a URL so we can control where it appears in layout.

**Comment by mattstratton** [comment]
I'm suspecting this is just a defect in the oembed-all library.

**Comment by aaronblythe** [comment]
I imagine there may be an issue in the oembed-all library, however this is fairly simple to work around in hugo #510 until that is patched.  The pull request would preserve interface to where the user would not have to figure out more than they need to, they simply would list the URL in the markdown for the speakerdeck setting, same as they would if the oembed-all library just worked.

I also now see what you mean about how shortcodes work in hugo, they are for the content not the html templates.
